### PR TITLE
storage: refactor store | The obvious starting blockchain and closing the database

### PR DESCRIPTION
**Pull request by volekerb**
Did a small pair programming with @im-kulikov during Friday and this PR is the first to step further.

### CODE REVIEW

File: pkg/core/blockchain.go [hunk 1]
```diff
@@ -154,7 +154,13 @@ func (bc *Blockchain) init() error {
 
 func (bc *Blockchain) run(ctx context.Context) {
 	persistTimer := time.NewTimer(persistInterval)
-	defer persistTimer.Stop()
+	defer func() {
+		persistTimer.Stop()
+		err := bc.Store.Close()
+		if err != nil {
```
Comment (im-kulikov) [inline pkg/core/blockchain.go]: what you think?

```go
if err := bc.Store.Close(); err != nil {
  log.Warnf("failed to close db: %s", err)
}
```

Comment (volekerb) [reply pkg/core/blockchain.go]: done

**Comment by roman-khimov** [comment]
I think these can be squashed as there is absolutely no point in having `Context` being passed and not used. And yeah, these are good.

**Comment by volekerb** [comment]
Squashed @roman-khimov

File: pkg/core/storage/memory_store.go [hunk 2]
```diff
@@ -76,6 +76,14 @@ func (s *MemoryStore) Batch() Batch {
 	}
 }
 
+// Close implements Store interface and clears up memory.
+func (s *MemoryStore) Close() error {
+	s.Lock()
+	s.mem = make(map[string][]byte)
```
[the remark this answers is not in the document]
Comment (im-kulikov) [reply pkg/core/storage/memory_store.go]: It will panics.. after this

[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/core/storage/memory_store.go]: And it's a feature to me, because you're not supposed to use `Store` after `Close()`.

**Pull request merged**